### DISCUSSION TITLE
[typescript-operations] Any way to automatically add a field to queries if it exists in the schema?

### DISCUSSION THREAD

**Discussion by threehams** [dotansimha/graphql-code-generator#4656] [General] [answered]
With the existing options on `typescript-operations`, is there a way to always add a field if it's available? Specifically, I'm wondering about `id`, since it's required for cache normalization and very easy to forget.

We're currently using a lint rule to require it, but two issues with that:
- wow, it's tedious to fix / add every time, especially when using fragments
- it now makes the `id` required on the type (it would be great, though not required, if the field were not part of the type unless it were explicitly added).

If not, would this be generally useful to add? I can contribute if we can agree on an option name.

**Comment by dotansimha** [comment, marked as answer]
Hi @threehams 
Codegen doesn't do schema/operations manipulations or that level. You can do that with a custom document loader: https://graphql-code-generator.com/docs/getting-started/documents-field#custom-document-loader
